src/Orchard.Cms.Web/Modules/Orchard.OpenId/Controllers/AdminController.cs

```diff
@@ -0,0 +1,330 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.Localization;
+using Microsoft.Extensions.Localization;
+using OpenIddict.Core;
+using Orchard.DisplayManagement;
+using Orchard.DisplayManagement.Notify;
+using Orchard.Navigation;
+using Orchard.OpenId.Models;
+using Orchard.OpenId.Services;
+using Orchard.OpenId.ViewModels;
+using Orchard.Security.Services;
+using Orchard.Settings;
+
+namespace Orchard.OpenId.Controllers
+{
+    [Authorize]
```

Comment:
FWIW, this hidden "is this controller an admin controller?" sorcery is a bit weird... I'd personally prefer an explicit model :smile: